src/wolfscp.c

```diff
@@ -2025,7 +2161,7 @@ int wsScpSendCallback(WOLFSSH* ssh, int state, const char* peerRequest,
             }
 
             ret = (word32)WFREAD(buf, 1, bufSz, sendCtx->fp);
-            if ((ret < 0) || (fileOffset + ret == *totalFileSz)) {
+            if ((ret <= 0) || (fileOffset + ret == *totalFileSz)) {
```

Comment:
changes behavior for test after read. Now also when file read returns 0 it is considered an error.